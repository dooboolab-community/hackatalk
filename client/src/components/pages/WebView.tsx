import React, {FC} from 'react';
import {RouteProp, useRoute} from '@react-navigation/core';

import {RootStackParamList} from '../navigations/RootStackNavigator';
import {WebView} from 'react-native-webview';
import styled from 'styled-components/native';

const Container = styled.SafeAreaView`
  flex: 1;
  background-color: ${({theme}) => theme.background};
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

const Page: FC = () => {
  const {
    params: {uri},
  } = useRoute<RouteProp<RootStackParamList, 'WebView'>>();

  return (
    <Container>
      <WebView source={{uri}} />
    </Container>
  );
};

export default Page;
