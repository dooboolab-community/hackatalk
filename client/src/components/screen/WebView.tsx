import React, { ReactElement } from 'react';
import { RootStackNavigationProps, RootStackParamList } from '../navigation/RootStackNavigator';

import { RouteProp } from '@react-navigation/core';
import { WebView } from 'react-native-webview';
import styled from 'styled-components/native';

const Container = styled.SafeAreaView`
  flex: 1;
  background-color: ${({ theme }): string => theme.background};
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

interface Props {
  navigation: RootStackNavigationProps<'WebView'>;
  route: RouteProp<RootStackParamList, 'WebView'>;
}

function Page(props: Props): ReactElement {
  const { route: { params: { uri } } } = props;

  return (
    <Container>
      <WebView source={{ uri }} />
    </Container>
  );
}

export default Page;
