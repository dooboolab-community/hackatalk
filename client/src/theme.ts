import type {DoobooTheme} from 'dooboo-ui';

/* eslint-disable */
export const colors = {
  brand: '#1E6EFA',
  paleBlue05: '#E2F2FF',
  paleBlue10: '#8BC9FF',
  green10: '#A1EEDB',
  green30: '#00D4AB',
  green50: '#28DB98',
  gray05: '#F3F9FF',
  success: '33ff2f',
  danger: '#ff002e',
  warning: '#f2df0f',
  info: '#3a74e7',
  light: '#ededed',
  dark: '#151C22',
  messagePrimary: '#8BC9FF',
  messageSecondary: '#E2F2FF',
  apple: '#151E22',
  google: '#E04238',
  facebook: '#345997',
};
/* eslint-enable */

export const light = {
  background: 'white',
  paper: '#F2F5F6',
  disabled: '#C4C4C4',
  line: '#gray05',
  text: '#000',
  textLight: '#1B1B1B',
  textContrast: 'white',
  placeholder: '#6D6D6D',
  header: colors.brand,
  card: 'white',
  active: colors.brand,
  primary: colors.brand,
  primaryLight: colors.paleBlue10,
  secondary: colors.green10,
  button: colors.brand,
};

export type Theme = typeof light & DoobooTheme;

export const dark = {
  background: '#000',
  paper: colors.dark,
  disabled: '#515151',
  line: '#000',
  text: 'white',
  textLight: '#D3D3D3',
  textContrast: '#000',
  placeholder: '#6D6D6D',
  header: '#000',
  card: colors.dark,
  active: colors.paleBlue10,
  primary: '#fff',
  primaryLight: '#151A25',
  secondary: colors.green30,
  button: colors.paleBlue10,
};
