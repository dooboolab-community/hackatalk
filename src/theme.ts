/* eslint-disable */
export const colors = {
  light: '#ffffff',
  dark: '#151A25',
  backgroundDark: '#151E22',
  deepDark: '#000000',
  blue_80: '#2F38C7',
  blue_70: '#1E6EFA',
  blue_60: '#217DFE',
  paleBlue_10: '#8bc9ff',
  paleBlue_05: '#e2f2ff',
  green_30: '#00D4AB',
  green_10: '#A1EEDB',
  gray_80: '#676c7a',
  gray_70: '#545966',
  gray_60: '#676c7a',
  gray_50: '#8f94a3',
  gray_40: '#AFB4C3',
  gray_30: '#d3d8e8',
  gray_05: '#f3f9ff',
  apple: '#151E22',
  google: '#E04238',
  facebook: '#345997',
};
/* eslint-enable */

export const light = {
  background: colors.light,
  backgroundDark: colors.paleBlue_05,
  itemBackground: colors.light,
  btnPrimary: colors.blue_70,
  btnPrimaryFont: colors.light,
  btnPrimaryLight: colors.light,
  btnPrimaryLightFont: colors.blue_70,
  textDisabled: colors.gray_50,
  btnDisabled: colors.gray_30,
  fontColor: colors.dark,
  fontSubColor: colors.blue_70,
  labelColor: colors.dark,
  tintColor: colors.green_30,
  lineColor: colors.gray_30,
  indicatorColor: colors.blue_70,
  inactiveColor: colors.gray_60,
  primary: colors.blue_70,
  primaryLight: colors.paleBlue_10,
  headerFont: colors.light,
  searchBackground: colors.gray_05,
  messageBackground: colors.light,
  status: colors.green_10,
  placeholder: colors.gray_60,
  focused: colors.blue_70,
  placeholderFocused: colors.blue_70,
  socialBtn: colors.paleBlue_05,
  link: colors.blue_70,
  myMessageBackground: colors.paleBlue_10,
  myMessageText: colors.dark,
  peerMessageBackground: colors.paleBlue_05,
  peerMessageText: colors.dark,
  modalBtnBackground: colors.light,
  modalBtnPrimaryFont: colors.blue_80,
  modalBtnFont: colors.gray_60,
  appleIcon: colors.apple,
  appleText: colors.apple,
  appleBackground: colors.light,
  facebookIcon: colors.light,
  facebookText: colors.light,
  facebookBackground: colors.facebook,
  googleIcon: colors.light,
  googleText: colors.light,
  googleBackground: colors.google,
  errorTitle: colors.gray_60,
  errorBody: colors.gray_50,
  fab: colors.blue_70,
};
export type Theme = typeof light;
export const dark = {
  background: colors.deepDark,
  backgroundDark: colors.deepDark,
  itemBackground: colors.dark,
  btnPrimary: colors.paleBlue_10,
  btnPrimaryFont: colors.deepDark,
  btnPrimaryLight: colors.dark,
  btnPrimaryLightFont: colors.paleBlue_10,
  textDisabled: colors.gray_80,
  btnDisabled: colors.gray_40,
  fontColor: colors.light,
  fontSubColor: colors.paleBlue_10,
  labelColor: colors.light,
  tintColor: colors.green_10,
  lineColor: colors.light,
  indicatorColor: colors.paleBlue_10,
  inactiveColor: colors.gray_50,
  primary: colors.deepDark,
  primaryLight: colors.dark,
  headerFont: colors.light,
  searchBackground: colors.gray_80,
  messageBackground: colors.dark,
  status: colors.green_30,
  placeholder: colors.gray_50,
  focused: '#ffffff',
  placeholderFocused: colors.paleBlue_10,
  socialBtn: colors.gray_05,
  link: colors.paleBlue_05,
  myMessageBackground: colors.deepDark,
  myMessageText: colors.light,
  peerMessageBackground: colors.paleBlue_05,
  peerMessageText: colors.dark,
  modalBtnBackground: colors.gray_80,
  modalBtnPrimaryFont: colors.paleBlue_10,
  modalBtnFont: colors.gray_40,
  appleIcon: colors.apple,
  appleText: colors.apple,
  appleBackground: colors.light,
  facebookIcon: colors.facebook,
  facebookText: colors.dark,
  facebookBackground: colors.light,
  googleIcon: colors.google,
  googleText: colors.dark,
  googleBackground: colors.light,
  errorTitle: colors.paleBlue_10,
  errorBody: colors.gray_40,
  fab: colors.paleBlue_10,
};
