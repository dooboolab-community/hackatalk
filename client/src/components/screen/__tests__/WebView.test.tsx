import 'react-native';

import * as React from 'react';

import { NavigationContainer } from '@react-navigation/native';
import { ThemeProvider } from '@dooboo-ui/theme';
import WebView from '../WebView';
import { createStackNavigator } from '@react-navigation/stack';
import { render } from '@testing-library/react-native';

type ParamList = {
  WebView: {
    uri: string;
  };
};

const Stack = createStackNavigator<ParamList>();

const Container: React.FC = () => {
  return (
    <ThemeProvider>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen
            name="WebView"
            component={WebView}
            initialParams={{
              uri: '',
            }}
          />
        </Stack.Navigator>
      </NavigationContainer>
    </ThemeProvider>
  );
};

const component = <Container />;

describe('[WebView] screen', () => {
  it('renders without crashing', () => {
    const json = render(component).toJSON();

    expect(json).toMatchSnapshot();
  });
});
