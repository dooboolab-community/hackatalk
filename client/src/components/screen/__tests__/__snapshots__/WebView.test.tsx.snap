// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[WebView] screen renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          false,
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                WebView
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTSafeAreaView
                      emulateUnlessSupported={true}
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#FFFFFF",
                            "flexBasis": 0,
                            "flexDirection": "row",
                            "flexGrow": 1,
                            "flexShrink": 1,
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "flex": 1,
                              "overflow": "hidden",
                            },
                            undefined,
                          ]
                        }
                      >
                        <RNCWebView
                          cacheEnabled={true}
                          injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
                          injectedJavaScriptForMainFrameOnly={true}
                          javaScriptEnabled={true}
                          messagingEnabled={false}
                          onContentProcessDidTerminate={[Function]}
                          onHttpError={[Function]}
                          onLoadingError={[Function]}
                          onLoadingFinish={[Function]}
                          onLoadingProgress={[Function]}
                          onLoadingStart={[Function]}
                          onMessage={[Function]}
                          onShouldStartLoadWithRequest={[Function]}
                          source={
                            Object {
                              "uri": "",
                            }
                          }
                          style={
                            Array [
                              Object {
                                "flex": 1,
                                "overflow": "hidden",
                              },
                              Object {
                                "backgroundColor": "#ffffff",
                              },
                              undefined,
                            ]
                          }
                          useSharedProcessPool={true}
                        />
                      </View>
                    </RCTSafeAreaView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
